### ISSUE TITLE
Improve godoc on `tea.Program.Start()`

### ISSUE THREAD

**Issue by NiloCK** [charmbracelet/bubbletea#299]
https://github.com/charmbracelet/bubbletea/blob/14e58aa1f92fa9bd756d9e739ce63a39e98a39ac/tea.go#L584

This is an important function, which will be called by every beginner user in every program.

- "Ignores the final model." is not a complete sentence, and I don't understand it anyhow.
- There is no indication that `Start` is blocking. I scratched my head quite a bit before replacing this:

```
if err := game.Start(); err != nil {
	fmt.Printf("error starting server game: %v", err)
	os.Exit(1)
}

// why isn't my connection receiving game updates and forwarding them?
for {
	bf := make([]byte, 5120)
		n, err := conn.Read(bf)
	if err != nil {
		panic(err)
	}

	var gs c4.GameState
	json.Unmarshal(bf[:n], &gs)
	game.Send(gs)
}
```

with this:

```
go func() {
	for {
		bf := make([]byte, 5120)

		n, err := conn.Read(bf)
		if err != nil {
			panic(err)
		}
		var gs c4.GameState
		json.Unmarshal(bf[:n], &gs)

		game.Send(gs)
	}
}()

if err := game.Start(); err != nil {
	fmt.Printf("error starting server game: %v", err)
	os.Exit(1)
}
```

See the godoc on [exec.Run](https://pkg.go.dev/os/exec#Cmd.Run) and [exec.Start](https://pkg.go.dev/os/exec#Cmd.Start) for examples of well communicated blocking / nonblocking.

**Comment by muesli** [comment]
I agree, the naming is a bit unfortunate indeed and calling it `Program.Run` could have avoided this irritation.

Two things we can do, as I see it:

1. Explicitly document the blocking nature of Start in the godocs.
2. Introduce `Program.Run` and update the examples & documentation. Deprecate `Start` and have it become an alias of `Run` for the time being.

The first one is an obvious and easy fix, not sure about the second option. Might be the right thing to do tho.

**Comment by inkel** [comment]
I took a stab at updating the docs in #479.

**Comment by inkel** [comment]
I for one think that would be a good idea, however, it is a quite disruptive change, so maybe it should be planned carefully 🤔 

Right now the _statu quo_ is using `Start`, and if newcomers already have problems with it because of confusing documentation then I'd start there, and slowly move towards implementing `Run`; all existing documentation, YouTube tutorials, etc, use `Start` so introducing the change now will be even more confusing IMO.

**Comment by muesli** [comment]
> I for one think that would be a good idea, however, it is a quite disruptive change, so maybe it should be planned carefully thinking

It's a change, but it doesn't have to be disruptive. We strive for backwards compatibility, and aliasing `Start` would ensure we don't break any existing code.

> Right now the _statu quo_ is using `Start`, and if newcomers already have problems with it because of confusing documentation then I'd start there, and slowly move towards implementing `Run`; all existing documentation, YouTube tutorials, etc, use `Start` so introducing the change now will be even more confusing IMO.

Agreed, #479 is probably a good start. On the other hand delaying such decisions just make the eventual change worse. The sooner we mend this, the better.

**Comment by muesli** [comment]
Thinking about it some more, we should probably simply remove the `...ReturningModel` function eventually. Omitting the returned model is something the caller can already do, with fewer characters to type.

**Comment by inkel** [comment]
I wasn't even aware of the existence of that method to be honest, and I couldn't find a use case for it in my limited experience. Perhaps the final model could be exposed via a `FinalModel() Model` method?

**Comment by muesli** [comment]
> I wasn't even aware of the existence of that method to be honest, and I couldn't find a use case for it in my limited experience. Perhaps the final model could be exposed via a `FinalModel() Model` method?

I think having a newly added `Run` method always return it is the straight forward solution. The caller can still decide to omit it like:

```go
_, err := p.Run()
```

**Comment by inkel** [comment]
Oh yes, that's definitely better.

**Comment by meowgorithm** [comment]
Yep, that all makes sense. Let's do it.

**Comment by muesli** [comment]
Closed by #479 and #509. Thanks for the ticket!
